lib/ModelMixins/AutoRefreshingMixin.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  computed,
+  IReactionDisposer,
+  onBecomeObserved,
+  onBecomeUnobserved,
+  reaction
+} from "mobx";
+import { now } from "mobx-utils";
+import Constructor from "../Core/Constructor";
+import Model from "../Models/Model";
+import MappableTraits from "../Traits/MappableTraits";
+
+type AutoRefreshing = Model<MappableTraits>;
+
+export default function AutoRefreshingMixin<
+  T extends Constructor<AutoRefreshing>
+>(Base: T) {
+  abstract class AutoRefreshingMixin extends Base {
+    _autoRefreshDisposer: IReactionDisposer | undefined;
+
+    /* Return the interval in seconds to poll for updates. */
+    abstract refreshInterval: number | undefined;
+
+    /* Call hook for refreshing the item */
+    abstract refreshData(): void;
+
+    constructor(...args: any[]) {
+      super(...args);
+      // We should only poll when our map items have consumers
+      onBecomeObserved(this, "mapItems", () => {
+        this._autoRefreshDisposer = reaction(
+          () => this._pollingTimer,
+          () => {
+            this.refreshData();
+          }
+        );
+      });
+      onBecomeUnobserved(this, "mapItems", () => {
+        if (
+          this._autoRefreshDisposer !== undefined &&
+          this._autoRefreshDisposer !== null
+        ) {
+          this._autoRefreshDisposer();
+        }
+      });
+    }
+
+    @computed
+    protected get _pollingTimer(): number | undefined {
+      if (this.refreshInterval !== null && this.refreshInterval !== undefined) {
+        return now(this.refreshInterval * 1000);
+      }
```

Comment:
IMO it's good practice to make the `return undefined` explicit when that is what's intended. Otherwise it looks like an accident that the function can return undefined.